web/assignments_controller.go

```diff
@@ -60,3 +60,46 @@ func (ac *AssignmentsController) Show(c *gin.Context) {
 		c.JSON(200, as)
 	}
 }
+
+// CreateSnapshot will always return the Assignment ID of a snapshot. It may return all of the snapshot immediately,
+// or if the details take time to compute the results of the snapshot will be retrievable with the assignment’s XID.
+// Example:
+//  "/assignments/:AID/snapshots"
+func (ac *AssignmentsController) CreateSnapshot(c *gin.Context) {
+	id := c.Param("AID")
+
+	if j, err := ac.App.Store.FindJob(id); err == storm.ErrNotFound {
+		c.JSON(404, gin.H{
+			"errors": []string{"Job not found"},
+		})
+	} else if err != nil {
+		c.JSON(500, gin.H{
+			"errors": []string{err.Error()},
+		})
+	} else if jr, err := startJob(j, ac.App.Store, models.JSON{}); err != nil {
```

Comment:
Could you hit this edge case with a unit test? I think the easiest way would be to create a job spec with a endAt which has already passed, and then creating a snapshot for it.